source/unified-test-format/unified-test-format.rst

```diff
@@ -0,0 +1,2397 @@
+  top-level `runOn`_. This would mean that a test runner cannot immediately skip
+  an entire file based on evaluation of the top-level `runOn`_; however, it also
+  means that humans editing a test file can no longer trust that the top-level
+  `runOn`_ applies to all tests in the file.
+
+During review, it was asked if a test runner could enforce that `test.runOn`_ is
+only more restrictive (and never more permissive) than the top-level `runOn`_.
+While this is technically feasible, it would add considerable complexity to a
+test runner to make such comparisons.
+
+
+Mixing event types in expectedEvent
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Can different event types (e.g. command monitoring, SDAM) appear within the
+``events`` array within `test.expectedEvents`_? To date, no specs (including
+SDAM) expect events multiple types of events in the same test. `observeEvents`_
+does allow tests to filter events per client, so a test could conceivably expect
+mixed types (e.g. CommandStartedEvent and ServerDescriptionChangedEvent).
+
+Perhaps the underlying question is: when collecting observed events on a client,
+should the test runner keep them in one ordered list or group them into separate
+lists (e.g. one list for command monitoring events and another for SDAM events)?
+
+
+Should failPoint support a read preference?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The `failPoint`_ operation currently uses a primary read preference. Should it
+solicit a ``readPreference`` argument to target nodes other than the primary?
+To date, no spec needs this behavior and this change could easily be deferred to
+a future minor version of the test format.
+
+
+Representing options in operation.arguments
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Existing spec tests are inconsistent in how they represent optional
+parameters for operations. In some cases, optional parameters appear directly
+in `operation.arguments`_ alongside required parameters (e.g. CRUD). In other
+cases, they are nested under an ``options`` key (e.g. Sessions).
+
+Most specs *are* consistent about how API methods are defined in spec documents.
+The CRUD spec was on the first documents to define API methods and can be
+credited with introducing the ``options`` parameter (e.g.
+``options: Optional<UpdateOptions>``); however, this was likely done for
+readability of the spec (avoiding very long method declarations and/or embedding
+``Optional`` syntax therein). Additionally, documenting options in a separate
+type (e.g. UpdateOptions) allows the declarations to be shared and reused across
+multiple methods.
+
+That said, this syntax is in no way prescriptive for language implementations.
+While some drivers to model options as a struct/object, others solicit them
+alongside required parameters (e.g. Python's keyword/named arguments).
+
+Should this spec require that optional parameters be nested under an ``options``
```

Comment:
I favor sticking to the API definition in the CRUD spec (i.e. nesting optional params under ``options`` like the CRUD spec pseudocode suggests) for consistency across our spec documents. Though it does negatively impact readability, I think in this instance, consistency might be more valuable?

Reply:
I'm fine with @isabelatkinson suggestion on this point. I was unaware flattening these structs could introduce error.

Reply:
Since `session` is ambiguous and would require manual handling in any driver, we agreed to _not_ nest things under `options` and also rename `arguments` to `parameters` to remove any impression that the arguments are ordered.

Reply:
Revised "Entity Test Operations" to explain that `options` nesting should never be used.

@reggi: I did not rename `arguments` to `parameters`. After giving it some more thought, I considered that `arguments` is used in nearly every existing spec test format and a rename would likely cause more confusion. If you still feel strongly, perhaps we can defer that to final review and see how the leads feel.

Reply:
Resolving, as @isabelatkinson followed up in another thread about her preference for `arguments`.